test/e2e/network/dns_common.go

```diff
@@ -373,17 +372,10 @@ func createDNSPod(namespace, wheezyProbeCmd, jessieProbeCmd, podHostName, servic
 	dnsPod := e2epod.NewAgnhostPod(namespace, podName, volumes, mounts, nil, "test-webserver")
 	dnsPod.Spec.Containers[0].Name = "webserver"
 
-	querier := e2epod.NewAgnhostContainer("querier", mounts, nil, wheezyProbeCmd)
+	querier := e2epod.NewAgnhostContainer("querier", mounts, nil, probeCmd)
 	querier.Command = []string{"sh", "-c"}
 
-	jessieQuerier := v1.Container{
```

Comment:
the problem I see here is that we are loosing coverage with the libc implementation, right now we have `muslc` and `libc` covered.

We should not completely remove it and instead use one image that has debian or other libc implementation

Reply:
You're not wrong.  It looks like agnhost is alpine:

```
ldd `which dig`
	/lib/ld-musl-x86_64.so.1 (0x7fa21cb68000)
	libdns-9.16.27.so => /usr/lib/libdns-9.16.27.so (0x7fa21c975000)
        ...
```

Thast said, if we establish an image just for DNS testing which is libc based, someone has to perpetually build new versions of it.

I wonder if we can talk $someone into building agnhost on both alpine and debian bases?

Reply:
just replace s/jessie/debian/ for names and s/debian:jessie/debian:stable-slim/ for images

Reply:
let me do the building of the image

Reply:
> we are losing coverage with the libc implementation

Haha, no, we lost the relevant coverage _years_ ago. (But it doesn't matter.)

The timeline here is:
1. Wheezy has a broken DNS implementation which gives ndots-like behavior even if you don't specify ndots. (I think? That's what it seems like from the old discussion.)
2. Kubernetes pre-1.0 assumed that behavior.
3. Jessie fixed that behavior, breaking Kubernetes (#10161).
4. Kubelet was fixed to set `ndots:5` in resolv.conf, fixing Kubernetes under jessie. (#10266)
5. At this time, the e2e test images were all still based on wheezy, so the DNS e2e tests were modified to also test against special jessie-based images (since the wheezy-based tests would have passed with or without the `ndots` fix). It's not clear why it tested against both rather than _only_ testing against jessie, but [Tim commented](https://github.com/kubernetes/kubernetes/pull/10266#issuecomment-115103614) "We can reduce the test case back to one when we do a new dnsutils container" so it was clearly intended to be temporary.
6. ... but apparently everyone forgot about that, and mistakenly assumed that the point of the jessie image was that we wanted to test against jessie specifically, rather than that we had just needed something newer than wheezy.

Anyway, while it may happen to be the case that the current version of this test tests two different libcs, that is not the _intent_ of the case (and I don't think there is any reason to believe that libc and musl behave differently here anyway).

Reply:
If we build agnhost for alpine and debian, I think it would be OK to test both - these tests are fast and cheap, and musl HAS been a problem in the past (not wrt ndots but other things) and we only figured it out when other people reported bugs.

I don't love the idea of having a test image whose sole purpose is DNS, because we are expected to support it (the original bug behind this PR)

Reply:
> Haha, no, we lost the relevant coverage years ago. (But it doesn't matter.)

@danwinship jessie image is using libc, isn't it?

>  believe that libc and musl behave differently here anyway).

not a lot, but there is precedent https://github.com/kubernetes/kubernetes/issues/112135

Reply:
OK... so, the syntax and semantics of `/etc/resolv.conf` are not sufficiently well defined that we can assume that all software in all containers will interpret it identically. So we need to have an e2e test for "kubelet writes out a resolv.conf that is compatible with the resolver libraries of languages/runtimes we expect people to use in pods". And the easiest way to test that something is interpreting resolv.conf correctly is to just have it resolve things, and the only thing we know for sure that pods will be able to resolve is the Kubernetes DNS stuff. So here we are.

But we don't need to test multiple resolvers in _every_ DNS test. We should split out the alternative-runtime DNS tests into a single separate test case, to make it easier to distinguish "your cluster DNS is broken" failures from "kubelet's resolv.conf generation doesn't work with some languages/runtimes" failures.

And maybe we should test more resolv.conf options, to be able to catch things like #112135?

Reply:
right know, by chance, we have this libc/muslc more or less covered .. not a lot but at least the basic is covered 

> And maybe we should test more resolv.conf options, to be able to catch things like https://github.com/kubernetes/kubernetes/issues/112135?

yeah, we can do a way more for testing, specially on e2e ... but it requires time  ... I've tried to use e2e testing for providing opportunities for people to contribute, but if people is not able to progress alone with minimum help then it steals time from the ones that don't have time ... and the ones that get their code merged leave once they got their reward of having a commit in the repo